hadoop-ozone/ozonefs-hadoop3-client/pom.xml

```diff
@@ -0,0 +1,118 @@
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License. See accompanying LICENSE file.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.apache.ozone</groupId>
+    <artifactId>ozone</artifactId>
+    <version>1.3.0-SNAPSHOT</version>
+  </parent>
+<!--
+  This is called "ozone-filesystem-hadoop3-client" to correspond with
+  the shaded hadoop jar that it works with:
+  "hadoop-client-api.jar", (as opposed to the unshaded hadoop jar:
+  "hadoop-common.jar")
+-->
+  <artifactId>ozone-filesystem-hadoop3-client</artifactId>
+  <name>Apache Ozone FS Hadoop shaded 3.x compatibility</name>
+  <packaging>jar</packaging>
+  <version>1.3.0-SNAPSHOT</version>
+  <dependencies>
+    <dependency>
+      <groupId>org.apache.ozone</groupId>
+      <artifactId>ozone-filesystem-hadoop3</artifactId>
+      <optional>true</optional>
+    </dependency>
+  </dependencies>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>include-dependencies</id>
+            <phase>prepare-package</phase>
+            <goals>
+              <goal>unpack</goal>
+            </goals>
+            <configuration>
+              <excludes>META-INF/versions/**/*.*</excludes>
+              <artifactItems>
+                <artifactItem>
+                  <groupId>org.apache.ozone</groupId>
+                  <artifactId>ozone-filesystem-shaded</artifactId>
+                  <version>${project.version}</version>
+                </artifactItem>
+              </artifactItems>
+              <outputDirectory>target/classes</outputDirectory>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+        <executions>
+          <execution>
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <transformers>
+                <transformer
+                        implementation="org.apache.maven.plugins.shade.resource.DontIncludeResourceTransformer">
+                  <resources>
+                    <resource>META-INF/BC1024KE.DSA</resource>
+                    <resource>META-INF/BC2048KE.DSA</resource>
+                    <resource>META-INF/BC1024KE.SF</resource>
+                    <resource>META-INF/BC2048KE.SF</resource>
```

Comment:
just curious~ what do BC1024KE.DSA and BC2048KE.SF mean?

Reply:
My understanding is that is used when shading jars to prevent the changes from causing security exceptions.  We use it here as well: https://github.com/apache/ozone/blob/48e3b49aaad361fdb8b082cbeb957f7b897e4c68/hadoop-ozone/ozonefs-shaded/pom.xml#L94-L103

That is where I got it from.

Reply:
ohh!! thanks!!